tests/providers/useBeforeAll.spec.ts

```diff
@@ -31,6 +31,14 @@ describe('upstream providers can satisfy requirements of downstream providers',
   it('test', noop)
 })
 
+describe('can use multiple providers in a single call', () => {
+  const p1 = provider<{}, NoContext>().before(noop).done()
+  const p2 = provider<{}, NoContext>().before(noop).done()
+  kont().useBeforeAll(p1, p2).done()
```

Comment:
Have `p2` depend on context from `p1` showing that the context typing works not just across the chain but across a group

Reply:
Yeah, that didn't work. The typing will be a bit trickier I think.

Reply:
Yep :)